src/globus_cli/commands/collection/update.py

```diff
@@ -151,11 +152,12 @@ def collection_update(
     """
     Update a Mapped or Guest Collection
     """
-    if sharing_restrict_paths is not None:
-        if not isinstance(sharing_restrict_paths, dict):
-            raise click.UsageError(
-                "--sharing-restrict-paths may not contain non-object JSON data"
-            )
+    if isinstance(sharing_restrict_paths, ParsedJSONData) and not isinstance(
+        sharing_restrict_paths.data, dict
+    ):
+        raise click.UsageError(
+            "--sharing-restrict-paths may not contain non-object JSON data"
```

Comment:
I recommend converting this from "may not contain non-object JSON data" to "must be a JSON object". I think that this will be a clearer statement of the requirements.

This suggestion applies to all locations that raise a UsageError based on the `.data` attribute.

Note that updating this will affect all of the tests that rely on the current double-negative texts.

Reply:
Yeah, I like that much better. I'll run through and make the change in a jiffy.